app/utilities/protocol_importers/step_description_builder.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module ProtocolImporters
+  class StepDescriptionBuilder
+    def self.generate(step_json)
+      return '' unless step_json[:description]
+
+      html_description = ''
+      html_description = "<p>#{remove_html(step_json[:description][:body])}</p>" if step_json[:description][:body]
+
+      # Add components from JSON
+      html_description += step_json[:description][:components]&.inject('') do |html_string, component|
+        sanitized_component = component.except(:type)
+        sanitized_component[:body] = remove_html(component[:body]) if component[:body]
+
+        html_string + ApplicationController
+                      .renderer
+                      .render(template: "templates/protocols_import/step_description/#{component[:type]}",
```

Comment:
Aren't many of those components too specific for protocolsIO to be placed in the above folder? Do you think `protocols_io` folder would be appropriate here?

Reply:
We were discussed with Matjaz about components. As I understood him, those components are pretty standard components for different protocols (reagent, temperature, duration, etc...) and we can expect similar components from other endpoints. 
Because we are building description here for "our" `Step.rb` from already normalized data I would say appropriate folder is this. But I agree, currently, our and their structure looks similar.